ejb3/src/main/java/org/jboss/as/ejb3/subsystem/ApplicationSecurityDomainDefinition.java

```diff
@@ -236,4 +239,62 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
             return null;
         };
     }
+
+    private static class BooleanCapabilityReferenceRecorder implements CapabilityReferenceRecorder {
```

Comment:
Would this class be better in WildFly Core so it can be shared?  I think there was a duplicate issue raised for the Undertow subsystem so we know at least two will need a solution for this.

Reply:
I wasn't sure because the usage is scarce and imo this class is a bit hacky. But I can push it to core as well.

Reply:
Lets see what others reviewing think, but if we know at least two subsystems need the same solution something common may be preferred.

Although one thing we may keep in mind is at some point we will be removing these application-security-domain resources but we are not quite at that point yet.

Reply:
I'm doubtful about putting this in the core 'controller' project. It is shared across two things but I don't think it's a great config style and I'm concerned about promoting it. We need to update Galleon's wildfly plugin to understand that 'true' means there's the requirement.  That's a necessary but pretty low priority thing.

In any case I don't think adding a shared thing should block this; it can be separate.

Reply:
Ok that makes sense @bstansberry TBH I am also thinking about the steps needed to remove these application-security-domain resources so if this is the only use then we just need to make sure any replacement doesn't trigger the same kind of issue to solve.